src/folding/nova/decider.rs

```diff
@@ -529,59 +533,31 @@ pub mod tests {
         test_relaxed_r1cs_gadget(circuit);
     }
 
-    // circuit that has the number of constraints specified in the `n_constraints` parameter. Note
-    // that the generated circuit will have very sparse matrices, so the resulting constraints
-    // number of the RelaxedR1CS gadget must take that into account.
-    struct CustomTestCircuit<F: PrimeField> {
-        _f: PhantomData<F>,
-        pub n_constraints: usize,
-        pub x: F,
-        pub y: F,
-    }
+    // CustomTestCircuit is a circuit that has the number of constraints specified in the
+    // `n_constraints` parameter. Note that the generated circuit will have very sparse matrices.
+    pub struct CustomTestCircuit<F: PrimeField>(WrapperCircuit<F, CustomFCircuit<F>>);
```

Comment:
Does this still follow the constraints of the `CubicCircuit`? If so, we can unify the structs and just add a `Default` impl which has the needed constraints and a constructor function that allows to pass constraint numbers.

Reply:
This `CustomTestCircuit` does not relate to the `CubicCircuit`.
This `CustomTestCircuit` is the `CustomFCircuit` (the circuit that implements the `FCircuit` trait, which allows to select the number of constraints that it will have), but since here we are not interested into using the `FCircuit` but into haveing a circuit that implements the `ConstraintSynthesizer` trait, we use the `WrapperCircuit` to wrap the `CustomFCircuit` and expose its constraints generation.

But I just realized that this circuit is never used (it was in a earlier version of this branch, my fault)  so I've removed it ([here the diff](https://github.com/privacy-scaling-explorations/folding-schemes/compare/55fa83bf66648f1c4fb1709662fbc0b871b4ffc1..b9bcfc28f66fcb4103ef5965883401fd488b1ac0)).